src/components/Map/index.tsx

```diff
@@ -0,0 +1,190 @@
+import * as React from 'react';
+import { GoogleMap, LoadScript, Marker } from '@react-google-maps/api';
+import { DefaultProps } from '../../common-types';
+import { Box } from '../Box';
+
+export interface UIMarker {
+	id: string | number;
+	lat: number;
+	lng: number;
+	title?: string;
+	icon?: string;
+	click?: () => void;
+}
+
+// Coordinates of Seattle, avoid having the map centered in the middle of the sea
+const DEFAULT_LATLNG = {
+	lat: 47.6062,
+	lng: -122.3321,
+};
+const DEFAULT_ZOOM_LEVEL = 3;
+const DETAIL_ZOOM_LEVEL = 12;
+
+const defaultMapOptions: google.maps.MapOptions = {
+	controlSize: 32,
+	scrollwheel: false,
+	disableDefaultUI: true,
+	zoomControl: true,
+	mapTypeControl: true,
+	mapTypeControlOptions: {
+		mapTypeIds: ['roadmap', 'hybrid'],
+	},
+	styles: [
+		{
+			elementType: 'all',
+			featureType: 'all',
+			stylers: [{ saturation: -50 }],
+		},
+		{
+			elementType: 'all',
+			featureType: 'administrative.country',
+			stylers: [{ gamma: 2 }],
+		},
+		{
+			elementType: 'all',
+			featureType: 'administrative.locality',
+			stylers: [{ gamma: 1.5 }],
+		},
+	],
+};
+
+// Return map bounds based on list of markers
+const getMapBounds = (markers: UIMarker[]) => {
+	const bounds = new google.maps.LatLngBounds();
+
+	markers.forEach((marker) => {
+		bounds.extend(
+			new google.maps.LatLng(
+				isNaN(marker.lat) ? 0 : marker.lat,
+				isNaN(marker.lng) ? 0 : marker.lng,
+			),
+		);
+	});
+	return bounds;
+};
+
+// Fit map to its bounds after the api is loaded
+const apiIsLoaded = (map: google.maps.Map, markers: UIMarker[]) => {
```

Comment:
nit-pick: maybe rename this to `onApiLoad` to make it clear that it's a callback rather than a method that returns a boolean indicating if the API is loaded or not.